suite/tests/linux/rseq.c

```diff
@@ -1141,10 +1151,9 @@ kernel_xfer_event(void *drcontext, const dr_kernel_xfer_info_t *info)
 #        error Unsupported arch
 #    endif
 #    ifdef DEBUG /* See above: special code in core/ is DEBUG-only> */
-        /* Check that the interrupted PC for the true abort case is *prior* to the
-         * committing store.
-         */
-        assert(info->source_mcontext->pc == (app_pc)test_rseq_native_abort_pre_commit);
+        /* Check that the interrupted PC for the true abort case is the handler. */
+        assert(!in_rseq_migration_test ||
+               info->source_mcontext->pc == (app_pc)test_rseq_native_abort_handler);
 #    endif
     }
```

Comment:
For !in_rseq_migration_test also, can we check info->source_mcontext->pc == (app_pc)test_rseq_native_abort_handler?

Reply:
No, each test has its own abort handler.  We'd have to add a global label exported from the asm for each test and then have a big || here or have another global saying which test it was...doesn't seem worth it.

Reply:
Or a global var pointing at which handler should be expected...but then have to write it from the asm, or again export it.  IMHO the check for one test is enough.

Reply:
Added a comment on this.